src/clr-angular/utils/focus/key-focus/key-focus.ts

```diff
@@ -108,7 +108,7 @@ export class ClrKeyFocus {
 
   private get currentItem() {
     if (this._current >= this.focusableItems.length) {
-      return null;
+      this._current = 0;
```

Comment:
Why should the focus move to the first item? This might make sense if you are on the first item and it is removed, but if you are on the last item and it is removed I would think focus should move to the new last focusable item? 

Your demo shows the latter behavior, so I would expect focus to move to the `this._current = this.focusableItems.length - 1`. 

If any item in the list is removed except the last item, the current index remains the same and items shift over so that shouldn't be an issue.

Reply:
Thanks for the review.
I have updated the request with the corresponding changes.